### ISSUE TITLE
lint fails on tool dependency packages of just tool_dependencies.xml

### ISSUE THREAD

**Issue by peterjc** [galaxyproject/planemo#341]
e.g. using https://github.com/peterjc/galaxy_blast as an example,

``` bash
$  planemo --version
planemo, version 0.18.1.dev0
$ ls -a packages/package_blast_plus_2_2_31
.  ..  .shed.yml  tool_dependencies.xml
$ planemo lint --report_level all --fail_level error packages/package_blast_plus_2_2_31 ; echo "Return value $?"
Return value 2
```

The same happens if recursing:

```
$ planemo lint --report_level all --fail_level error -r packages/ ; echo "Return value $?"
Return value 2
```

I expected `planemo lint` to at least check `tool_dependencies.xml` was a valid XML file, and return 0 (success), even if nothing more rigorous was done.

**Comment by peterjc** [comment]
Motivating use case: https://github.com/biopython/galaxy_packages/commit/91c4d21e959ac19abbb270ff0bd9897ce31ad2f0

**Comment by jmchilton** [comment]
We explicitly added the ``2`` return test case so planemo lint wouldn't just pass if it failed to find any tools. Even if the xml is well-formed you wouldn't want planemo to pass if the outer tag was wrong for instance - maybe ``<tools>``. Lots of IDEs auto-populate the closing tag, so it wouldn't necessarily require the same package typo twice.

**Comment by peterjc** [comment]
So this boils down to user error in part: I should be using ``planemo shed_lint`` here to cover any kind of Tool Shed repository, so not just a tool-containing TS entry, but equally a just a dependency definition, or a workflow.

With hindsight, ``planemo lint`` could have been called ``planemo tool_lint``? Probably too late to change it?

I suggest we refocus this issue on better user-feedback - in addition to returning the special value 2, could ``planemo lint`` print something like "No tools found" to stdout/stderr?

**Comment by jmchilton** [comment]
Agreed completely, ``No tools found`` would be a great message to print.

I don't regret it being called ``lint`` instead of ``tool_lint``. Same with ``test`` instead of ``tool_test``. Planemo should be first and foremost about tools - this shed stuff is just a fun distraction :smiley:.

**Comment by peterjc** [comment]
Great.

P.S. I fixed https://github.com/peterjc/galaxy_blast/commit/13caf90673ac12e2c53e52c1f495194c8cf0a2d8 while my other repositories were already correctly using ``planemo shed_lint`` with TravisCI.

**Comment by jmchilton** [comment]
tool_dependencies.xml are long dead so I'm going to close this - thanks for the report! -John
